bin/webpack.js

```diff
@@ -161,10 +170,16 @@ function processOptions(options) {
 	}
 
 	var firstOptions = Array.isArray(options) ? (options[0] || {}) : options;
-
-	if(typeof options.stats === "boolean" || typeof options.stats === "string") {
+	if(hasPreset(options)) {
 		var statsPresetToOptions = require("../lib/Stats.js").presetToOptions;
-		options.stats = statsPresetToOptions(options.stats);
+
+		if(Array.isArray(options)) {
+			options = options.forEach(function(singleOption) {
+				singleOption.stats = statsPresetToOptions(singleOption.stats);
+			});
+		} else {
+			options.stats = statsPresetToOptions(options.stats);
+		}
 	}
 
 	var outputOptions = Object.create(options.stats || firstOptions.stats || {});
```

Comment:
Or if you are looking to not have the logic in `webpack.js`:

```js
		if(Array.isArray(options)) {
			options = options.forEach(function(singleOption) {
				singleOption.stats = statsPresetToOptions(singleOption.stats);
			});
		} else {
			options.stats = statsPresetToOptions(options.stats);
		}
```
I think this much enough could be extracted and put into `Stats.js` as a static/object method.